### PR TITLE
feat: added authentication

### PR DESCRIPTION
**Pull request by SanjayJatti**
- Created AuthContext and AuthReducer
- Added login and sign up page
auth-video-libarary

### CODE REVIEW

File: src/Pages/Authentication/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  const { email, password } = authState.userInfo;
+  const navigate = useNavigate();
+
+  const logInHandler = async (e, emailId, userPassword) => {
+    e.preventDefault();
+    try {
+      const response = await axios.post(`/api/auth/login`, {
+        email: emailId,
+        password: userPassword,
+      });
```
Comment (parthgabani777) [inline src/Pages/Authentication/Login.jsx:25]: You can make auth-services for making API calls

**Review by parthgabani777 — COMMENTED** [review]
In this netlify hosted link I am getting error 500 when trying to login/signup, make sure you are passing data in the right format in the request body & make sure you are setting up "REACT_APP_JWT_SECRET" environment variable in netlify

**Review by narharikale — APPROVED** [review]
code looks clean 
good to go 
and add .env key in env variable

**Comment by SanjayJatti** [comment]
> In this netlify hosted link I am getting error 500 when trying to login/signup, make sure you are passing data in the right format in the request body & make sure you are setting up "REACT_APP_JWT_SECRET" environment variable in netlify

Will look into it👍

**Pull request merged**